src/audio/eq_iir/eq_iir.c

```diff
@@ -783,13 +783,13 @@ static int eq_iir_cmd(struct comp_dev *dev, int cmd, void *data,
 
 static int eq_iir_trigger(struct comp_dev *dev, int cmd)
 {
-	struct comp_data *cd = comp_get_drvdata(dev);
+	struct comp_data *cd __unused = comp_get_drvdata(dev);
 
 	comp_info(dev, "eq_iir_trigger()");
 
-	if (cmd == COMP_TRIGGER_START || cmd == COMP_TRIGGER_RELEASE)
-		assert(cd->eq_iir_func);
-
+	if (cmd == COMP_TRIGGER_START || cmd == COMP_TRIGGER_RELEASE) {
+		; assert(cd->eq_iir_func);
```

Comment:
what is this semicolon here for?

Reply:
the extra semicolon indeed looks a bit whacky

Reply:
mmh, don't understand, sorry. That warning appears when `assert()` is expanded to nothing, right? So, without that additional semicolon you'd have a single semicolon there. With the additional one you have two of them. Not sure it improves anything ;-) Was it the compiler complaining? I think it isn't very rare to have code like
```
while (do_something())
	;
```
with an obvious empty loop body. Maybe you shouldn't add those superfluous braces?

Reply:
@marc-hb How did you get the empty statement warning? I tried with various empty/NULL definitions of assert (with your PR), but couldn't get a warning. 

@lgirdwood I'd like to resolve this issue before merging. But no other concerns with the PR.

Reply:
> what is this semicolon here for?

I finally remembered:

```
./scripts/checkpatch.pl -g HEAD
WARNING: braces {} are not necessary for single statement blocks
#67: FILE: src/audio/eq_iir/eq_iir.c:790:
+	if (cmd == COMP_TRIGGER_START || cmd == COMP_TRIGGER_RELEASE) {
 		assert(cd->eq_iir_func);
+	}
```
Anyway #5771 will supersede this hunk.